backend/dataall/modules/dataset_sharing/services/share_processors/lakeformation_process_share.py

```diff
@@ -0,0 +1,241 @@
+                    new_state = shared_item_SM.run_transition(ShareItemActions.Failure.value)
+                    shared_item_SM.update_state_single_item(self.session, share_item, new_state)
+                    success = False
+
+                    self.handle_share_failure(table=table, error=e)
+
+        return success
+
+    def process_revoked_shares(self) -> bool:
+        """
+        1) For each revoked table:
+            a) Update its status to REVOKE_IN_PROGRESS with Action Start
+            b) Check if table exists on glue catalog raise error if not and flag share item status to failed
+            c) Check if resource link table exists in target account
+            d) Check if the table is shared in other share requests to this target account
+            e) If c is True (resource link table exists), revoke permission to principals to resource link table
+            f) If c is True (resource link table exists), revoke permission to principals to table (and for QS Group if no other shares present for table)
+            g) If c is True and (old-share or (new-share and d is True, no other shares of this table)) then delete resource link table
+            g) If d is True (no other shares of this table with target), revoke permissions to target account to the original table
+            h) update share item status to REVOKE_SUCCESSFUL with Action Success
+        2) Check if there are existing_shared_tables for this dataset with target environment
+        3) If no existing_shared_tables, delete shared database
+
+        Returns
+        -------
+        True if share is revoked successfully
+        False if revoke fails
+        """
+        log.info(
+            '##### Starting Revoking tables #######'
+        )
+        success = True
+        for table in self.revoked_tables:
+            share_item = ShareObjectRepository.find_sharable_item(
+                self.session, self.share.shareUri, table.tableUri
+            )
+
+            revoked_item_SM = ShareItemSM(ShareItemStatus.Revoke_Approved.value)
+            new_state = revoked_item_SM.run_transition(ShareObjectActions.Start.value)
+            revoked_item_SM.update_state_single_item(self.session, share_item, new_state)
+
+            try:
+                log.info(f'Revoking access to table: {table.GlueTableName} ')
+                self.check_table_exists_in_source_database(share_item, table)
+
+                resource_link_table_exists = self.check_resource_link_table_exists_in_target_database(table)
+                other_table_shares_in_env = True if ShareObjectRepository.other_approved_share_item_table_exists(
+                    self.session,
+                    self.target_environment.environmentUri,
+                    share_item.itemUri,
+                    share_item.shareItemUri
+                ) else False
+
+                if resource_link_table_exists:
+                    self.revoke_principals_permissions_to_resource_link_table(table)
```

Comment:
do we not need to check `other_table_shares_in_env` to determine if we need to remove QS principal (if dashboards enabled) before revoking here as well?

Reply:
Found a potential error with removing QS Permissions - check out testing outcomes for more detail